### PR TITLE
Add xilinx-runtime recipe

### PR DESCRIPTION
**Pull request by bryanloz-xilinx**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by bryanloz-xilinx** [comment]
I am @bryanloz-xilinx and I am willing to be listed as a maintainer.
In the future I would like to recruit more helpers.

**Comment by bryanloz-xilinx** [comment]
@conda-forge/help-c-cpp this is ready for review.

I am hoping for some feedback.

I am currently using the git_url method for the source, and hoping to get an exception to the rule, for the very first recipe. Upon our next formal software release, I can update the recipe to use the url and the SHA.

For the first iteration, I intend not to support Windows or OSX. This package also has no need of CUDA. I have skipped the build in the meta.yaml. Hopefully this is the right way to do it.

The recipe built fine in my local mamba build. However, it is failing due to overlinking errors in the pipeline. I would love some feedback on that error. I'll paste error at bottom of this comment.

Lastly, this software is well verified on gcc9, however I am not specifying the gcc version in this recipe, because I want to allow for maximum flexibility. By default, I see that conda-forge uses gcc12. What is the standard practice here? Should I force gcc9?

```
conda_build.exceptions.OverLinkingError: overlinking check failed 
["  ERROR (xilinx-runtime,bin/unwrapped/xclbinutil): .. but ['openssl'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)", "  ERROR (xilinx-runtime,bin/unwrapped/xclbinutil): .. but ['boost-cpp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)", "  ERROR (xilinx-runtime,bin/unwrapped/xclbinutil): .. but ['boost-cpp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)", "  ERROR (xilinx-runtime,bin/xbflash2): .. but ['boost-cpp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)", "  ERROR (xilinx-runtime,lib/libxrt_core.so.2.14.0): .. but ['boost-cpp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)", "  ERROR (xilinx-runtime,bin/unwrapped/xbutil2): .. but ['boost-cpp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)", "  ERROR (xilinx-runtime,bin/unwrapped/xbutil2): .. but ['boost-cpp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)", "  ERROR (xilinx-runtime,lib/libxrt_coreutil.so.2.14.0): .. but ['boost-cpp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)", "  ERROR (xilinx-runtime,lib/libxrt++.so.2.14.0): .. but ['boost-cpp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)", "  ERROR (xilinx-runtime,lib/libxrt_swemu.so.2.14.0): .. but ['boost-cpp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)", "  ERROR (xilinx-runtime,lib/libxrt_swemu.so.2.14.0): .. but ['libprotobuf'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)", "  ERROR (xilinx-runtime,bin/unwrapped/xbmgmt2): .. but ['boost-cpp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)", "  ERROR (xilinx-runtime,bin/unwrapped/xbmgmt2): .. but ['boost-cpp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)", "  ERROR (xilinx-runtime,lib/libxrt_hwemu.so.2.14.0): .. but ['boost-cpp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)", "  ERROR (xilinx-runtime,lib/libxrt_hwemu.so.2.14.0): .. but ['libprotobuf'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)", "  ERROR (xilinx-runtime,lib/libxma2api.so.2.14.0): .. but ['boost-cpp'] not in reqs/run, (i.e. it is overlinking) (likely) or a missing dependency (less likely)"]
##[error]Bash exited with code '1'.
```

**Comment by bryanloz-xilinx** [comment]
@conda-forge/help-c-cpp this is ready for review.

File: recipes/xilinx-runtime/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    - python
+    - boost
+    - boost-cpp
+    - openssl
+    - libprotobuf
+    - rapidjson
+    - libuuid
+    - libdrm
+    - clhpp
+    - libcurl
```
Comment (xhochy) [inline recipes/xilinx-runtime/meta.yaml:51]: These should be added automatically by their run_exports. Can you remove the marked ones?

File: recipes/xilinx-runtime/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/Xilinx/XRT/archive/refs/tags/{{ quarter }}.{{ version }}.tar.gz
+  sha256: d73d8366f6f6d66bfdcb67e72135f559a4d95644c5416c27dea79c235953fd20
+  patches:
+    - latest.patch
```
Comment (xhochy) [inline recipes/xilinx-runtime/meta.yaml:13]: Do you really need this huge patch? I'm reluctant to merge something which has a 150k+ loc patch.

File: recipes/xilinx-runtime/meta.yaml (hunk 1 above)
Comment (bryanloz-xilinx) [reply recipes/xilinx-runtime/meta.yaml:51]: @xhochy if I did not specify openssl and boost-cpp conda-build would error out, saying that some of our binary executables were underlinking/overlinking... i.e. xclbinutil uses boost-cpp but boost-cpp was not found in run requirements.

File: recipes/xilinx-runtime/meta.yaml (hunk 2 above)
Comment (bryanloz-xilinx) [reply recipes/xilinx-runtime/meta.yaml:13]: @xhochy I was worried about this. Basically, the large patch is moving our most recent release, to head. Then I apply the smaller patch, to make things work with conda. The head of our repository was much easier to get working with conda, and hence the second patch is much simpler.

However, I think I agree with you. It is not quite fair of me to ask that you revision control my 150k loc patch.

I'll optimize this.

File: recipes/xilinx-runtime/meta.yaml (hunk 1 above)
Comment (xhochy) [reply recipes/xilinx-runtime/meta.yaml:51]: It should be sufficient to have them in host only. Can you give this a try and then let us look together at the error in CI?

**Comment by bryanloz-xilinx** [comment]
Closing in favor of #19337

**Pull request closed without merging**